fs_overlay/opt/certs_manager/certs_manager.rb

```diff
@@ -29,13 +37,23 @@ def setup
       config_domains(NAConfig.domains)
       Nginx.setup
 
-      Nginx.start
+      if initial
+        Nginx.start
+      else
+        Nginx.reload
+      end
 
       ensure_signed(NAConfig.domains, true)
 
-      Nginx.stop
+      if initial
+        Nginx.stop
+      else
+        Nginx.reload
+      end
+    end
+    if initial
+      sleep 1 # Give Nginx some time to shutdown
```

Comment:
Yes, we can; I just tried to keep as much of the structure as possible.

I guess it would improve readabilty; the downside is that the lock would be held one more second. I'm fine with both; your call.